java/src/main/native/src/RmmJni.cpp

```diff
@@ -349,19 +369,50 @@ class java_event_handler_memory_resource final : public device_memory_resource {
                                  on_dealloc_threshold_method, "onDeallocThreshold", total_after);
   }
 
-  bool supports_get_mem_info() const noexcept override { return resource->supports_get_mem_info(); }
+  void on_allocated_callback(std::size_t num_bytes, rmm::cuda_stream_view stream) {
```

Comment:
It's a bit odd that these methods are part of this class, and this class shouldn't have to risk looking up Java methods it has no intentions of calling.  This logic belongs in the derived class because it's specific to that class.

Reply:
I was wondering if we should have a class that is the "event handler wrapper". All it does is expose to the C++ JNI code the methods of the handler, and so it can get passed to the production java handler or the debug java handler in order to call those functions. Does this sound reasonable?